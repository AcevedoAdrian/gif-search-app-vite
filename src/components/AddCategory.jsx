import { useState } from 'react';

export const AddCategory = ({onNewCategory}) => {

  const [intputValue, setIntputValue] = useState("");
  
  const handleInputChange = ({ target }) => {
    setIntputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault(); 
    if(intputValue.trim().length <= 1 ) return;
    // onCategories(categories =>( [...categories, intputValue] ));
    onNewCategory(intputValue.trim());
    setIntputValue('');
  };


  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Buscar giff"
        value={intputValue}
        onChange={(e) => handleInputChange(e)}
      />
    </form>
  );
};
