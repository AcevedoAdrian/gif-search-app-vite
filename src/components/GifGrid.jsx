import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';

import { GifItem } from './GifItem';

export const GifGrid = ({category}) => {

 const {images , isLoading} = useFetchGifs(category);

  return (
    <>
    {
      isLoading && <p>Cargando</p>
    }
      <h3>{category}</h3>
      <div className='card-grid'>
        {
          images.map(
            (image) =>(
              <GifItem
                key={image.id} 
                {...image}
              />
            )
          )
        }
      </div>
    </>

  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}