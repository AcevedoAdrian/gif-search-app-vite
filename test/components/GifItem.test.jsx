import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Prueba sobre el componente <GifItem />',()=>{

  const title = 'Lufy';
  const url   = 'https://one-piece/lufy.jpg'
  test('Evaluacion del snapshot', ()=>{
    const {container} = render(<GifItem  title={title} url={url}/>);
    expect(container).toMatchSnapshot();
    
  });

  test('debe de mostrar la imagen con el URL y en alt indicado', ()=>{
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url)
    const {src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('debe de mostrar el titulo en el componente',()=>{
     render(<GifItem title={title} url={url} />);
     expect(screen.getByText(title)).toBeTruthy();
  });
});