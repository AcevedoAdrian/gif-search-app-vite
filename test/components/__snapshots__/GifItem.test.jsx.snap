// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre el componente <GifItem /> Evaluacion del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Lufy"
      src="https://one-piece/lufy.jpg"
    />
    <p>
      Lufy
    </p>
  </div>
</div>
`;
